js/packages/components/chat/member-bar/MemberBar.tsx

```diff
@@ -0,0 +1,114 @@
+import LottieView from 'lottie-react-native'
+import React, { useCallback, useEffect, useState } from 'react'
+import { StyleSheet, TouchableOpacity, View } from 'react-native'
+
+import { useAppDispatch, useConversationMembers, useMessengerClient } from '@berty/hooks'
+import { useNavigation } from '@berty/navigation'
+import { getPeerFromMemberPK, PeerNetworkStatus } from '@berty/redux/reducers/messenger.reducer'
+
+import { Avatars } from './Avatars'
+import { Boosted } from './Boosted'
+import { MemberBarItem } from './interfaces'
+
+interface MemberBarProps {
+	convId: string
+}
+
+export const MemberBar: React.FC<MemberBarProps> = props => {
+	const navigation = useNavigation()
+	const [animationStep, setAnimationStep] = useState(0)
+	const messengerClient = useMessengerClient()
+	const members = useConversationMembers(props.convId).filter(members => !members.isMe)
+	const dispatch = useAppDispatch()
+
+	const [memberList, setMemberList] = useState<MemberBarItem[] | undefined>(undefined)
+
+	const handleMemberList = useCallback(() => {
+		const list: MemberBarItem[] = []
+
+		if (!messengerClient) {
+			return
+		}
+
+		members.forEach(async member => {
+			const peer = await dispatch(
+				getPeerFromMemberPK({ memberPK: member.publicKey, convPK: props.convId }),
+			)
+
+			if (peer.payload && list.length < 5) {
+				list.push({
+					networkStatus: peer.payload as PeerNetworkStatus,
+					publicKey: member.publicKey ?? undefined,
+				})
+			}
+		})
```

Comment:
it would be better to shorten the list before calling `forEach` instead of fetching the network status for the whole list and only pushing 5

I guess you do that to still have 5 elements in case some peer.payload is undefined?

In this case maybe use a `for (const member of members)`and break out of the loop after the list if full

Reply:
Nope we still 5 elements because the UI show only 5 peers in the bar but in this case too i think too that it's efficient to use `for (const member of members)` and break out of the loop when we have 5 elements